### PR TITLE
Some updates on Stream and Print class.

### PR DESCRIPTION
**Pull request by cmaglie**

### CODE REVIEW

File: cores/arduino/Print.h [hunk 1]
```diff
@@ -54,6 +57,10 @@ class Print
       return write((const uint8_t *)buffer, size);
     }
 
+    // default to zero, meaning "a single write may block"
+    // should be overriden by subclasses with buffering
+    virtual int availableForWrite() { return 0; }
```
Comment (sandeepmistry) [inline cores/arduino/Print.h]: Should we make `Uart:: availableForWrite()` virtual as well?

Comment (cmaglie) [inline cores/arduino/Print.h]: yes... good point...

Comment (sandeepmistry) [inline cores/arduino/Print.h]: Same for I2S, those are the only two spots in the repo.

Comment (sandeepmistry) [inline cores/arduino/Print.h]: Based on testing, adding the virtual doesn't seem to make a difference, we just needed to change the `size_t` to `int` in I2S.

**Pull request merged**